Assets/Scripts/SS3D/Systems/Tile/TileMap.cs

```diff
@@ -329,12 +350,21 @@ public void Clear()
 
             _chunks.Clear();
 
+            // Clear items list safely, checking for null references
+            var itemsToRemove = new List<PlacedItemObject>();
             foreach (PlacedItemObject item in _items)
             {
-                item.DestroySelf();
+                if (item != null && item.gameObject != null)
+                {
+                    item.DestroySelf();
+                }
+                itemsToRemove.Add(item);
+            }
+            
+            foreach (var item in itemsToRemove)
+            {
+                _items.Remove(item);
```

Comment:
```suggestion
                while (_items.Count > 0)
            {
                PlacedItemObject item = _items.First();
                if (item != null && item.gameObject != null)
                {
                    item.DestroySelf();
                }
                
                _items.RemoveAt(0);
            }
```
Also overwrite lines above, github doesn't allow to comment them